# HG changeset patch
# User Thorsten Meyer <thorsten.meyier@gmx.de>
# Date 1228715123 -3600
#      Mon Dec 08 06:45:23 2008 +0100
# Branch release-3-0-x
# Node ID 177611afa7c19f33819ae00785702ab7e0dc87ee
# Parent  03dab4d76975727221450a191ba49ab403148e8a
fix bug in cell string handling of unpack function

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2008-11-04  Thorsten Meyer <thorsten.meyier@gmx.de>
+
+        * miscellaneous/unpack.m: return directly after recursive handling
+          of cell-strings
+          
 2008-11-24  Ben Abbott <bpabbott@mac.com>
 
 	* plot/legend.m: Correct ording of legend labels.
 
 2008-03-20  Ben Abbott <bpabbott@mac.com>
 
 	* statistics/base/statistics.m: Calculate median and quantiles in
 	a manner consistent with method #7 used by GNU R.
diff --git a/scripts/miscellaneous/unpack.m b/scripts/miscellaneous/unpack.m
--- a/scripts/miscellaneous/unpack.m
+++ b/scripts/miscellaneous/unpack.m
@@ -92,16 +92,22 @@ function filelist = unpack (file, direct
 
   elseif (iscellstr (file))
     files = {};
     for i = 1:numel (file)
       tmpfiles = unpack (file{i}, directory);
       files = {files{:} tmpfiles{:}};
     endfor
 
+    ## Return output if requested.
+    if (nargout > 0)
+      filelist = files;
+    endif
+
+    return
   else
     error ("unpack: invalid input file class, %s", class(file));
   endif
 
   ## Instructions on what to do for any extension.
   ##
   ## The field names are the file extension without periods.
   ## The first cell is what is executed to unpack an archive verbosely.
